projects/element-ng/tabs-next/si-tab-next.component.ts

```diff
@@ -32,31 +30,41 @@ export class SiTabNextComponent extends SiTabNextBaseDirective implements OnDest
    * */
   override readonly active = model(false);
 
-  /** @internal */
-  override selectTab(retainFocus?: boolean): void {
+  protected selectTabByUser(): void {
     const tabs = this.tabset.tabPanels();
     const newTabIndex = this.index();
     const currentTabIndex = this.tabset.activeTabIndex();
     let continueWithSelection = newTabIndex !== currentTabIndex;
+    const currentTab = tabs[currentTabIndex];
 
-    if (continueWithSelection && currentTabIndex !== -1) {
-      const currentTab = tabs[currentTabIndex];
+    if (continueWithSelection && currentTab) {
       const deselectEvent = {
         target: currentTab,
         tabIndex: currentTabIndex,
         cancel: () => {
           continueWithSelection = false;
-          currentTab.active.set(true);
         }
       };
-
-      currentTab.active.set(false);
       this.tabset.deselect.emit(deselectEvent);
     }
 
     if (continueWithSelection) {
-      this.active.set(true);
-      super.selectTab(retainFocus);
+      this.selectTab();
     }
   }
+
+  /** @internal */
+  override selectTab(retainFocus?: boolean): void {
+    const tabs = this.tabset.tabPanels();
+    const currentTabIndex = this.tabset.activeTabIndex();
+    const currentTab = tabs[currentTabIndex];
+
+    currentTab?.deSelectTab();
+    this.active.set(true);
+    super.selectTab(retainFocus);
+  }
+
+  override deSelectTab(): void {
```

Comment:
Shouldn't this one be documented as `@ internal` or is it intended to be used outside of our code-base?

Reply:
I did not make my mind up about this fully. Select tab should be public in my opinion. So it would be logic to keep deSelectTab as well in the public.

I am also planning to write documentation/js-docs in a follow-up PR. I guess then I also need to decide on this.

Reply:
@spike-rabbit well, as long as you take care of the follow-up MR before v48, it's all fine with me. We should just avoid making such things public and then internal again after a few releases, if people start using it wrongly.